packages/sanity/src/core/tasks/hooks/useActivityLog.ts

```diff
@@ -65,10 +66,11 @@ export function useActivityLog(task: TaskDocument): {
     [transactionsUrl, token, publishedId],
   )
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, unused-imports/no-unused-vars
+  const handleFetchAndParse = useEffectEvent((rev: string) => fetchAndParse(task))
   useEffect(() => {
-    fetchAndParse(task)
     // Task is updated on every change, wait until the revision changes to update the activity log.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [fetchAndParse, task._rev])
+    handleFetchAndParse(task._rev)
```

Comment:
Sending the `task._rev` here is just to make it a valid `useEffect` dependency in the eyes of `eslint-plugin-react-hooks/exhaustive-deps` as well as the React Compiler.